lib/uri.js

```diff
@@ -118,23 +128,33 @@ URI.prototype.expand = function(params) {
                 if (segmentValue === undefined) {
                     if (segment.modifier) {
                         // Okay to end the URI here
-                        return new URI(res);
+                        uri = new URI(res);
+                        uri._pathMetadata = pathMetadata;
+                        return uri;
                     } else {
                         throw new Error('URI.expand: parameter ' + segment.name + ' not defined!');
                     }
                 }
             }
-            if (Array.isArray(segmentValue)) {
-                res = res.concat(segmentValue);
+
+            if (segment.modifier === '+') {
+                // Res will become a path array, so we must split path elements
+                var splitPath = ('' + segmentValue).split('/');
+                res = res.concat(splitPath);
+                for (var j = res.length - 1; j < res.length - 1 + splitPath.length; j++) {
```

Comment:
This `for` is rather confusing. What is going on here? You are assigning `pathMetadata[j]` for values `j, res.length - 1 <= j < splitPath.length + res.length - 1`, but the length of `res` has already been augmented for `splitPath.length` on the line above it. So, effectively, you are iterating for values `oldRes.len + splitPath.len - 1 <= j < oldRes.len + 2 * splitPath.len - 1`.